### ISSUE TITLE
Could you please tell me the implementation details of the experiment for the visualization in Figure 4?

### ISSUE THREAD

**Issue by LGRlgr6666** [luow23/INP-Former#45]
Figure 4. Visualization of the impact of INP coherence loss
Lc. (a) Input anomalous image and ground truth. (b) Distance
map without Lc. (c) Distance map with Lc. The distance map
is obtained by calculating the distance between the input features
and their nearest INP terms, as described in Eq. 2.

**Comment by luow23** [comment]
<img width="828" height="125" alt="Image" src="https://github.com/user-attachments/assets/9e9aab5f-73e7-4343-a749-498d897bb3c6" />Under the default setting, the shape of self.distance is （B, 28*28. During post-processing, it is reshaped to (B,28,28) for visualization.

<img width="713" height="54" alt="Image" src="https://github.com/user-attachments/assets/d420dac0-6e09-4ee2-ac1f-7debc5c7e79a" />

**Comment by LGRlgr6666** [comment]
Thank you very much for your careful answer. Your work is wonderful

**Comment by LGRlgr6666** [comment]
Do I just need to comment out this line of your code? 

“distance = prototype_map[i].view((28,28)).cpu().detach().numpy()”
Then the effects of the second and third lines as follows can be obtained about Figure 4

**Comment by LGRlgr6666** [comment]
What I mean is that I still don't specifically understand how to obtain the effect diagrams of the second row and the third one in Figure 4 from your experiment. If possible, I hope you can tell me the specific operations, and I would be very grateful.

**Comment by LGRlgr6666** [comment]
Can I run any of the test codes you provided?
such as python INP-Former_Multi_Class.py --dataset MVTec-AD --data_path ../mvtec_anomaly_detection --phase test

**Comment by luow23** [comment]
If you aim to achieve the effect shown in the second row of Figure 4, you can simply train the model without applying the gather loss.
